addons/analytic/models/analytic_plan.py

```diff
@@ -105,6 +107,13 @@ def _compute_children_count(self):
         for plan in self:
             plan.children_count = len(plan.children_ids)
 
+    @api.constrains('parent_id')
+    def _check_different_company_on_parent_plan(self):
+        if not self.parent_id:
+            return
+        if self.parent_id.company_id != self.company_id:
+            raise UserError(_('The parent and child analytic plans must belong to the same company. Please ensure that their company information is updated accordingly.'))
```

Comment:
What about setting `_check_company_auto = True` on the class and `check_company=True` on the `parent_id` declaration instead?